### PR TITLE
Feature/setfiles performance

### PR DESCRIPTION
**Pull request by ghost**
Could you take a looksie here? This is definitely not yet ready to merge, I just want feedback. Some DAPS functionality is probably broken right now on this branch.

Some other caveats are explained in the commit messages/inline comments.

The goal of this is to improve the performance of DAPS's file list functionality which are somewhat important for my Gedit build integration and for Tomas's vim integration they are too.

The tldr here is that for one specific book (DC-SLES-security), I got this time down from 8.5s to 6.1s on my notebook. Which is a somewhat humble start, I guess.

I have some more ideas on improving that time which I don't know how to realize yet:

1. I noticed that the first xsltproc call takes more than twice as long when called from within make (1.7s v/ 3.6s for DC-SLES-security, as measured with `time`). My (wholly uneducated) guess is that this might have something to do with disabled threading, but I did not see a `-j` parameter or similar anywhere ... Ideas?

2. Checking for profiling is really expensive (at least when running through make, but not so much when running xsltproc directly, interestingly) and we always do it for the whole document, even though we only need to do it for the things that are under `$rootid`. However, as the Makefile is set up right now, we only determine whether we have a rootid after the main XSLT has run. Any good ideas for moving that logic further up?

### CODE REVIEW

File: daps-xslt/profiling/check.profiling.xsl [hunk 1]
```diff
@@ -1,169 +1,59 @@
...
-   
+
 <xsl:include href="pi-attribute.xsl"/>
 
 <xsl:param name="provo.root"/>
 
 <xsl:template name="check.profiling">
   <xsl:param name="context" select="."/>
+  <xsl:param name="attributes" select="$profiling.attributes.enabled"/>
+  <xsl:param name="values" select="$profiling.values.merged"/>
```
Comment (tomschr) [inline daps-xslt/profiling/check.profiling.xsl]: [Travis](https://travis-ci.org/openSUSE/daps/builds/239516232?utm_source=github_status&utm_medium=notification) says:

```
make[2]: Entering directory `/home/travis/build/openSUSE/daps'
runtime error: file /home/travis/build/openSUSE/daps/daps-xslt/profiling/base-profile.xsl line 56 element call-template
Variable 'profiling.attributes.enabled' has not been declared.
[...]
runtime error: file /home/travis/build/openSUSE/daps/daps-xslt/profiling/base-profile.xsl line 56 element call-template
Variable 'profiling.values.merged' has not been declared.
```

Comment (ghost) [reply daps-xslt/profiling/check.profiling.xsl]: As I said, my changes so far were focused on getting `list-srcfiles` to be faster... All else, including profiling, I will have to fix later on.

**Comment by fsundermeyer** [comment]
For testing:

    #!/bin/bash
    #DDIR=/local/git/daps
    DDIR=/usr/share/daps
    #PROFILE="--stringparam profile.os sles --stringparam profile.arch \"x86_64;zseries;power;aarch64\""
    PROFILE=""

    xsltproc $PROFILE --stringparam rootid "book.sle.admin" \
     --stringparam xml.src.path "/local/git/doc-sle/xml/" \
     --stringparam mainfile "MAIN.SLEDS.xml" \
     ${DDIR}/daps-xslt/common/get-all-used-files.xsl \
     /local/git/doc-sle/xml/MAIN.SLEDS.xml | \
     xsltproc --stringparam filetype xml --stringparam rootid "book.sle.admin" \
     ${DDIR}/daps-xslt/common/extract-files-and-images.xsl -

**Comment by ghost** [comment]
@fsundermeyer I have the feeling I won't work on this again. Should we just close this and forget it ever existed?

**Comment by fsundermeyer** [comment]
Lets wait until after Hackweek. Maybe I have a chance to look at it again, if not, lets delete it...

**Comment by ghost** [comment]
Ok, let's close.

**Pull request closed without merging**